micronaut-langchain4j-core/src/main/java/io/micronaut/langchain4j/store/memory/chat/cassandra/CassandraChatMemoryStoreFactory.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2017-2025 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.langchain4j.store.memory.chat.cassandra;
+
+import dev.langchain4j.store.memory.chat.cassandra.CassandraChatMemoryStore;
+import io.micronaut.context.annotation.EachBean;
+import io.micronaut.context.annotation.Factory;
+import io.micronaut.context.annotation.Prototype;
+import io.micronaut.core.annotation.Internal;
+import io.micronaut.core.util.CollectionUtils;
+import io.micronaut.core.util.StringUtils;
+import jakarta.inject.Named;
+import jakarta.inject.Singleton;
+
+@Factory
+@Internal
+class CassandraChatMemoryStoreFactory {
+    @Named("cassandra")
+    @Prototype
+    CassandraChatMemoryStore.Builder createCassandraChatMemoryStoreBuilder(CassandraChatMemoryStoreConfiguration config) {
+        CassandraChatMemoryStore.Builder builder = CassandraChatMemoryStore.builder();
```

Comment:
it is likely better to use `@ConfigurationBuilder` here instead of replicating the properties inside `CassandraChatMemoryStoreConfiguration`

Reply:
[I changed to use `@ConfigurationBuilder` for cassandra](https://github.com/micronaut-projects/micronaut-langchain4j/pull/173/commits/6959d44b1a498660c21c5609c060d259d7fd9f46)